pkg/scheduler/framework/v1alpha1/interface.go

```diff
@@ -36,6 +36,9 @@ type NodeScoreList []int
 // PluginToNodeScoreMap declares a map from plugin name to its NodeScoreList.
 type PluginToNodeScoreMap map[string]NodeScoreList
 
+// NodeToStatusesMap declares map from node name to its statuses.
+type NodeToStatusesMap map[string][]*Status
```

Comment:
why a slice? this will be used to carry the status of a filtered node, and there will be only a single such status since we don't continue executing the filters if one returns unschedulable, so this slice will always be of size 1.

Reply:
Related to https://github.com/kubernetes/kubernetes/pull/78097/files#r309180696, will update this if we reach a consensus on whether to separate the status from predicate failure or not and how to do it.